uv/manager/manager.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python
+#
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""The MeasurementManager class, which conducts and reports on measurements
+which may be measured at varying intervals through training."""
+
+from typing import Dict, Iterable, List, Any, Type, Union, Callable
+from uv.reporter import AbstractReporter
+
+
+class MeasurementManager:
+  """Class for a MeasurementManager.  This manager is instantiated with a
+  particular reporter, which it will use to report whatever measurements is
+  conducts.
+
+  The MeasurementManager has a set of measurements---stored in a dictionary
+  (described below)---which are to be carried out at a certain frequency, which
+  is specific to each measurement.
+
+  Whenever the MeasurementManager's process() function is called, with a
+  step number provided as an argument, the MeasurementManager decides which
+  measurements are required at each step, carries them out, and reports them.
+
+  In order to carry out the measurement, the Manager has a state (Dict) at each
+  step, which contains all the information necessary to perform the measurement.
+  Part of this state is static (e.g., the training dataset), while part can
+  change at each step.  This dynamic state is passed along in the process() call.
+
+  Args:
+    static_state: Dict of variables which will be used to compute the measured
+                  values.  The static_state is fixed throughout time.
+    reporter: Reporter which will report the measurements.
+  """
+
+  def __init__(self, static_state: Dict[str, Any],
+               reporter: Type[AbstractReporter]):
```

Comment:
nice! do you need the `Type` wrapper?

Reply:
Good catch.  I had misunderstood what `Type` does.